registry/storage/driver/testdriver/testdriver.go

```diff
@@ -20,11 +22,48 @@ func (factory *testDriverFactory) Create(parameters map[string]interface{}) (sto
 	return New(), nil
 }
 
+// GetContentFunc is the type of GetContent
+type GetContentFunc func(ctx context.Context, path string) ([]byte, error)
+
+// PutContentFunc is the type of PutContent
+type PutContentFunc func(ctx context.Context, path string, content []byte) error
+
+// ReaderFunc is the type of Reader
+type ReaderFunc func(ctx context.Context, path string, offset int64) (io.ReadCloser, error)
+
+// WriterFunc is the type of Writer
+type WriterFunc func(ctx context.Context, path string, append bool) (storagedriver.FileWriter, error)
+
+// StatFunc is the type of Stat
+type StatFunc func(ctx context.Context, path string) (storagedriver.FileInfo, error)
+
+// ListFunc is the type of List
+type ListFunc func(ctx context.Context, path string) ([]string, error)
+
+// MoveFunc is the type of Move
+type MoveFunc func(ctx context.Context, sourcePath string, destPath string) error
+
+// DeleteFunc is the type of Delete
+type DeleteFunc func(ctx context.Context, path string) error
+
+// URLForFunc is the type of URLFor
+type URLForFunc func(ctx context.Context, path string, options map[string]interface{}) (string, error)
+
 // TestDriver is a StorageDriver for testing purposes. The Writer returned by this driver
 // simulates the case where Write operations are buffered. This causes the value returned by Size to lag
 // behind until Close (or Commit, or Cancel) is called.
 type TestDriver struct {
 	storagedriver.StorageDriver
+	// The below is to override the default GetContent on StorageDriver so that we can force error, etc. easily
```

Comment:
Thanks! I thought to make the mock driver generic enough so that other future tests can use it. I will follow your suggestion to only scope the driver for this new manifest test if that is preferred.